libs/FirmwareKit/source/FirmwareKit.cpp

```diff
@@ -10,7 +10,12 @@ using namespace leka;
 
 auto FirmwareKit::getCurrentVersion() -> FirmwareVersion
 {
+#if defined(FIRMWARE_VERSION)
```

Comment:
I'm not a big fan of macros, especially this deep in the implementation. 

my understanding is that `#if defined(FIRMWARE_VERSION)` is always true, even if the value is the empty string `""`

If I'm not mistaken, then the other path is never reached and can be removed.

what is the rational for keeping the `getCurrentVersionFromFile` part?

I'm really careful with these macro magic tricks. My take would be "force the firmware version everywhere, there is no reason to not define/use it and skip all the checks".

Reply:
> my understanding is that #if defined(FIRMWARE_VERSION) is always true, even if the value is the empty string ""

Not it is not, if it is the case I will not implement it this way.

But I get the point of the reliability of macro, so let me know what is your suggestion